doc/gettingStarted.md

```diff
@@ -0,0 +1,337 @@
+
+Make the following changes to the XYZ_Frog_EnumerateInstance method in mi_main.py:
+```
+def XYZ_Frog_EnumerateInstances (
+ context, nameSpace, className, propertySet, keysOnly):
++ frog1 = context.NewInstance ('XYZ_Frog')
++ frog1.SetValue ('Name', MI_String ('Fred'))
++ frog1.SetValue ('Weight', MI_Uint32 (55))
++ frog1.SetValue ('Color', MI_String ('Green'))
++ context.PostInstance (frog1)
++ frog2 = context.NewInstance ('XYZ_Frog')
++ frog2.SetValue ('Name', MI_String ('Sam'))
++ frog2.SetValue ('Weight', MI_Uint32 (65))
++ frog2.SetValue ('Color', MI_String ('Blue'))
++ context.PostInstance (frog2)
++ context.PostResult (MI_RESULT_OK)
+- context.PostResult (MI_RESULT_NOT_SUPPORTED)
+```
+
+### Registering the Provider:
+
+Next, register the provider as follows:
+```
+> ../../bin/omireg --Python XYZ_Frog
+Created /opt/omi/etc/omiregister/root-cimv2/XYZ_Frog.reg
+```
+
+The omniserver needs to reload the registry for this change to take effect.
+This can be acomplished by restarting the omiserver, or by running the following command:
+```
+> sudo ../../bin/omiserver -r
+../../bin/omiserver: refreshed server
+```
+
+### Testing the Provider:
+
+To test the provider, send an enumerate request to the provider as shown:
+```
+> ../../bin/omicli ei root/cimv2 XYZ_Frog
+instance of XYZ_Frog
+{
+ [Key] Name=Fred
+ Weight=55
+ Color=Green
+}
+instance of XYZ_Frog
+{
+ [Key] Name=Sam
+ Weight=65
+ Color=Blue
+}
+```
+
+
+## Going further:
+
+This provides a brief overview of the provider development process.
+The next section contains a deeper discussion.
+
+### Provider File Location and OMI Registry:
+
+This is a more technical explanation of how the OMI registry works to associate providers to OMI queries.
+This outlines how the OMI registry works, and how the OMI finds the libraries to execute.
+
+### OMI Register:
+
+The OMI Server associates Classes with library files through a collection of registry files.
+When the OMI Server starts, it reads the registry files and creates links between namespaces and classes,
+to the library files that handle them. The simplest registry files contain a value for "LIBRARY" and a value for "CLASS".
+The registry file associates a provider (LIBRARY) with the classes (CLASS) that it has handlers for.
+The registry files are stored in a predetermined location and have a ".reg" extension.
+Depending on which options the OMI Server was built with and particular environment variables,
+the registry files could be in a number of places. If OMI Server was installed with a package,
+the registry files are placed in "/etc/opt/omi/conf/omiregister/".
+If OMI Server was configured with the "--dev" option and built from source,
+the registry files are placed at "[omi-source-root]/omi/Unix/output/etc/omiregister/".
+There are a number of directories within the omiregister directory. Each of these subdirectories represents a namespace.
+And each ".reg" file represents a single provider.
```

Comment:
Might be better to split this in multiple paragraphs.

Reply:
Done!